### ISSUE TITLE
Duplicate risk check performed after recovering from #1497

### ISSUE THREAD

**Issue by ndegendogo** [corona-warn-app/cwa-app-ios#1505]
## Avoid duplicates

* [ ] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [ ] Bug is not already reported in another issue

## Describe the bug

A friend installed cwa 1.5.3 two days ago for his first time; next day updated to 1.6.0. Before his very first risk check he opened the app, but was affected by issue #1497. He pressed home button while error message was still visible, and waited for several hours.
Several hours later he followed the recommendations in the FAQ, stopped cwa (swipe), and started cwa again - this time successfull - and got the check. 
But ENF exposure log shows two attempts: first with 13 files (the newest missing), immediately followed by another loop (now all 14 files).

cwa 1.6.0 / iOS 14.2 / iPhone 12

## Expected behaviour
A single loop of all 14 files.

## Steps to reproduce the issue
Guessing a bit:
- have 14 files downloaded, but then a serious issue like #1497 blocks the check
- check is blocked for several hours
- oldest file expires
- cwa is started again, unblocks the pending check, and triggers another checking task (file download and check). 

So the unblocked check presents only 13 files, and the new check presents all 14 files.

Does this make sense? 🤪

See screenshot with timestamps and ENF log file.
Note : the check on 12 Nov was already with cwa 1.6.1

## Technical details

- cwa 1.6.0
- iOS Version: 14.2
- Device: iPhone 12

## Additional context

[ExposureChecks-2020-11-12.json.txt](https://github.com/corona-warn-app/cwa-app-ios/files/5533759/ExposureChecks-2020-11-12.json.txt)

![Screenshot 2020-11-12 at 23 06 04](https://user-images.githubusercontent.com/6118267/99007460-8ab42e00-2544-11eb-8bea-484546eff6fa.png)

---
Internal Tracking ID: EXPOSUREAPP-3711

**Comment by dsarkar** [comment]
Hi @ndegendogo, 

Thanks for posting this issue. It seems similar if not the same issue https://github.com/corona-warn-app/cwa-app-ios/issues/1475#issuecomment-724365359.

Best wishes,
DS

Corona-Warn-App Open Source Team

**Comment by ndegendogo** [comment]
@dsarkar well - I was aware of #1475, but decided for a new ticket because the error pattern is different. And only the devs can analyse if it has maybe a single root cause or not.
- #1475: a **single** error check with **corrupted** list of files (containing duplicate entries)
- this ticket: **two** error checks in close sequence

**Comment by dsarkar** [comment]
@ndegendogo, thanks for pointing out the difference. There is an internal JIRA ticket (Internal Tracking ID: EXPOSUREAPP-3711),  regarding risk calculation being triggered too often, sometimes within seconds. We will let know the developers also about this specific issue you published here.

**Comment by Ein-Tim** [comment]
Related PR: https://github.com/corona-warn-app/cwa-app-ios/pull/1506 (planed for 1.7.0)

**Comment by ndegendogo** [comment]
... and today I could watch it live.

As always, I firstly checked the ENF log. It showed **two** checks today, at 13:49 with 14 files (too **early**; yesterday was at 17:45). Second check at 17:49 (in time, but **0 files**!)

Then I opened cwa - it showed some error screen, I don't remember the exact message.
After ~2 seconds it started to download data for some time, then performed the check for some time. After this another download and another check.

ENF log file shows now **5 checks** in total. 17:49 now has 13 files (oldest missing). All other have full 14 files.

cwa 1.6.1(0) / iOS 14.2 / iPhone 8

**Comment by Ein-Tim** [comment]
Sometimes there are not only 2 checks performed at the same time but more, e.g. 3
![image](https://user-images.githubusercontent.com/67682506/99220463-08936600-27df-11eb-90b4-a884a5fdd664.jpeg)
I think this is also the root cause for the "ExposureDetectionIsAlreadyRunning"-Error.

**Comment by mss1010** [comment]
> Sometimes there are not only 2 checks performed at the same time but more, e.g. 3
> 
> ![image](https://user-images.githubusercontent.com/67682506/99220631-498b7a80-27df-11eb-8100-c011ee7fa5ba.jpeg)
> 
> I think this is also the root cause for the "ExposureDetectionIsAlreadyRunning"-Error. (#1497)

I also had the same behavior. **Three** checks when I opened the app manually **before** the 24-hour period finished.

**Comment by Ein-Tim** [comment]
@mss1010 

> before the 24-hour period finished.

This is at the moment (with 1.6.1) intended behavior (see https://github.com/corona-warn-app/cwa-app-ios/issues/1514#issuecomment-727602796)

**Comment by Ein-Tim** [comment]
I think this is solved with 1.7.1, or @ndegendogo

**Comment by ndegendogo** [comment]
> I think this is solved with 1.7.1

Yes, this one and a lot more.
1.7.1 is the best release since long. 🚀

Closing the ticket
